src/shared/azureagent/periodic.go

```diff
@@ -0,0 +1,77 @@
+package azureagent
+
+import (
+	"context"
+	"fmt"
+	"github.com/otterize/intents-operator/src/shared/errors"
+	"github.com/sirupsen/logrus"
+	"strings"
+)
+
+func (a *Agent) PeriodicTasks() {
+	ctx := context.TODO()
```

Comment:
Provide an external context for it, probably the one passed to `NewAzureAgent`. This is so that, if the context is canceled (eg by sigterm), the periodic tasks will gracefully terminate as well. 

You can do this: 
```
go wait.Forever(func() {agent.PeriodicTasks(ctx)}, 5*time.Hour)
```